erc721-lockable/contracts/base/ERC721Lockable.sol

```diff
@@ -0,0 +1,453 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Openzeppelin
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+// Interfaces
+import "../interface/IBunzz.sol";
+import "../interface/IERC721Lockable.sol";
+
+/**
+ * @title ERC721Lockable
+ * @notice ERC721 Token that can be locked for a certain period and cannot be transferred.
+ * This is designed for a non-escrow staking contract that comes later to lock a user's NFT
+ * while still letting them keep it in their wallet.
+ * This extension can ensure the security of user tokens during the staking period.
+ * If the nft lending protocol is compatible with this extension, the trouble caused by the NFT
+ * airdrop can be avoided, because the airdrop is still in the user's wallet
+ * @dev Implementation ERC721 Lockable Token
+ * @author kazunetakeda25
+ */
+abstract contract ERC721Lockable is ERC721, IERC721Lockable, IBunzz, Ownable {
+    // Mapping from token ID to unlock time
+    mapping(uint256 => uint256) public lockedTokens;
+
+    // Mapping from token ID to lock approved address
+    mapping(uint256 => address) private _lockApprovals;
+    // Mapping from owner to lock operator approvals
+    mapping(address => mapping(address => bool)) private _lockOperatorApprovals;
+
+    /**
+     * @dev Connect to other contracts
+     */
+    function connectToOtherContracts(address[] calldata _contracts)
+        public
+        override
+        onlyOwner
+    {}
```

Comment:
if you don't need this function, please remove this function and `IBunzz` interface